modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/TableDmlIntegrationTest.java

```diff
@@ -458,6 +458,45 @@ public void testMergeKeysConflict() {
             "Failed to MERGE some keys due to keys conflict");
     }
 
+    /**
+     * Ensure that DML operations fails with proper errors on non-existent table
+     */
+    @Test
+    public void testFailureOnNonExistentTable() {
+        assertThrows("INSERT INTO NON_EXISTENT_TABLE(ID, NAME) VALUES (1, 'Name')",
+            IgniteSQLException.class,
+            "Failed to validate query. From line 1, column 13 to line 1, column 30: " +
```

Comment:
Let's just check substring `"Object 'NON_EXISTENT_TABLE' not found"`, not full message (checking position looking weird, at least for UPDATE statement with column 1-55)

Reply:
Sure. Much better.